### PR TITLE
fix(test): update failing unit tests

### PR DESCRIPTION
**Pull request by jhonasiv**

### CODE REVIEW

**Comment by jhonasiv** [comment]
There are more tests that are failing, namely:

- On test/test_configurations.rb:
"extracts the configuratoin from the task and saves it": `raising Orocos::CORBAError: failed to marshal /ComplexStructure` on line 987.
"extracts the task's configuration and saves it to disk": `raising Orocos::CORBAError: failed to marshal /ComplexStructure` on line 928.

- On test_output_reader:
"should raise ComError if the remote end is dead and disconnected": nothing is being raised.

- On test_task:
"should raise CORBA::ComError when #operation has communicatoin errors": `raising Orocos::CORBA::TimeoutError` instead on line 125
"should allow to record its state trace when using extended state support": `expected the state to be RUNNING but it was RUNTIME_ERROR` on line 221. Apparently it stays as RUNTIME_ERROR after the first time it was raised.

- On test_nameservice:
"allows registering a task explicitely and updates it": `raising RuntimeError: Size of descriptions exceeds maximum allowed payload size: current 1643 vs. maximum 1300` on line 332

**Review by caioaamaral — APPROVED** [review]

File: test/test_port.rb [hunk 1]
```diff
@@ -105,11 +105,13 @@
         describe "validation of message size" do
             it "initializes data_size by the value returned by #max_marshalling_size f data_size is zero" do
                 flexmock(port).should_receive(:max_marshalling_size).and_return(10)
+                flexmock(Orocos::MQueue).should_receive(:validate_sizes?).and_return(false)
                 assert_equal Hash[transport: Orocos::TRANSPORT_MQ, size: 42, data_size: 10],
                     port.handle_mq_transport("input", transport: 0, size: 42, data_size: 0)
             end
             it "initializes data_size by the value returned by #max_marshalling_size f data_size is not given" do
                 flexmock(port).should_receive(:max_marshalling_size).and_return(10)
+                flexmock(Orocos::MQueue).should_receive(:validate_sizes?).and_return(false)
```
[the remark this answers is not in the document]
Comment (jhonasiv) [reply test/test_port.rb:114]: validate_sizes would overwrite the returned data_size when true

File: test/test_properties.rb [hunk 2]
```diff
@@ -150,10 +150,13 @@
     it "should be able to enumerate its attributes" do
         Orocos.run('process') do |process|
             t = process.task('Test')
-            assert_equal %w{att1 att2 att3}, t.attribute_names.sort
-            assert_equal %w{att1 att2 att3}, t.each_attribute.map(&:name).sort
-            %w{att1 att2 att3}.each do |name|
-                t.has_attribute?(name)
+            usual_attributes =
+                %w[CycleCounter IOCounter TimeOutCounter TriggerCounter TriggerOnStart]
+            task_attributes = %w[att1 att2 att3]
```
[the remark this answers is not in the document]
Comment (jhonasiv) [reply test/test_properties.rb:155]: rtt/rtt/TaskContext.cpp

[the remark this answers is not in the document]
Comment (caioaamaral) [reply test/test_properties.rb:155]: > rtt/rtt/TaskContext.cpp

I mean: `%w[att1 att2 att3]`

[the remark this answers is not in the document]
Comment (jhonasiv) [reply test/test_properties.rb:155]: It's in a orogen file specific to this test (see the `working_copy/process` folder inside of tests)

**Comment by jhonasiv** [comment]
Closing this with https://github.com/rock-core/tools-orocosrb/pull/149, since that PR is on top of this one and this one breaks tools/syskit. A fix for that is already included on https://github.com/rock-core/tools-syskit/pull/335, there is no point in having another PR when they would need to be merged together.

**Pull request closed without merging**